cupy/_core/_kernel.pyx

```diff
@@ -1584,21 +1613,18 @@ cdef class _Ops:
         return _Ops(tuple(ops_))
 
     cpdef _Op guess_routine(
-            self, str name, dict cache, list in_args, dtype, _Ops out_ops):
+            self, str name, dict cache, list in_args, tuple weaks, dtype,
+            _Ops out_ops):
         cdef _Ops ops_
         if dtype is None:
-            use_raw_value = _check_should_use_min_scalar(in_args)
```

Comment:
Not for me to decide.  If we wish to support both 1.x and 2.x (for the most part), it could make sense to keep the `_check_should_use_min_scalars` here.

As a very subtle point, NumPy still (usually) uses the same check as in `if all(weak): weak = [False] * len(weak)`.

(Which is arguably annoying, and only has an effect for weird integer inputs, though, so I think it is OK to ignore it.)

Reply:
Not for me to decide either. This PR so far labored under assumption that cupy 14 only supports numpy 2.0, per https://github.com/cupy/cupy/issues/8306#issuecomment-2076426835.
BTW, this implies that cupyx.scipy requires scipy 1.13